<a href="https://colab.research.google.com/github/ravindranaithani/week3.ipynb/blob/main/Pandas_Assignment_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Import Libraries
import pandas as pd
import numpy as np

In [ ]:
df = pd.read_csv("https://drive.google.com/uc?id=16D9FztGYzHiOXHzz5FiojqVbyOyV6zM2&export=download")
df.replace('?',np.nan,inplace=True)

In [ ]:
df.head()

Date  Year    Locality  Estimated Value  Sale Price       Property  \
0  2009-01-02  2009   Waterbury         111440.0    185000.0  Single Family   
1  2009-01-02  2009  Bridgeport         124670.0    150000.0     Two Family   
2  2009-01-02  2009   Waterbury          55720.0    140000.0  Single Family   
3  2009-01-02  2009  Bridgeport        4775276.0    272900.0  Single Family   
4  2009-01-02  2009  Bridgeport         112351.0    210000.0            NaN   

      Residential  num_rooms  num_bathrooms  carpet_area  property_tax_rate  \
0  Detached House          3              3        996.0           1.025953   
1          Duplex          4              3       1241.0           1.025953   
2  Detached House          3              2        910.0           1.025953   
3  Detached House          3              1        971.0           1.025953   
4  Detached House          3              2       1092.0           1.025953   

    Face  
0  South  
1  South  
2  South  
3   East  
4   East

#Q1 [Marks: 2] Select all the even indexed rows from the given dataset. What is the value in the 3rd indexed column of the 0th indexed row in the selected dataframe. (row/ column indexed from zero(0) in python/pandas)
- Please consider case sensitivity while giving your solution capital letter $\neq$ small letter
- If it is a string write without quotes in the solution.

Ans: 111440 (Range: 111439 - 111441)


In [ ]:
# Select all even-indexed rows
selected_df = df.iloc[::2]

# Find the value in the 3rd indexed column of the 0th indexed row in the selected DataFrame
value = selected_df.iloc[0, 3]

# Print the result
print("Value in the 3rd indexed column of the 0th indexed row:", value)

Value in the 3rd indexed column of the 0th indexed row: 111440.0


Example:

||column 0|column1|column2|column3|
|:-------|:-------|:------|:------|:------
|0|V00|V01|V02|V03     
|1|V10|V11|V12|V13
|2|V20|V21|V22|V23

- value in 1st column of the 2th row = V21


#Q2 [Marks: 2] Select all the even indexed rows from the given dataset. What is the value in the 3rd indexed column of the 332th indexed row in the selected dataframe. (row/ column indexed from zero(0))
- Please consider case sensitivity while giving your solution capital letter $\neq$ small letter
- If it is a string write without quotes in the solution.

Ans : 256200 (Range : 256199 - 256201)

In [ ]:
df.iloc[::2].reset_index(drop=True).iloc[332,3]

256200.0

#Q3 [Marks: 2] Select all the odd indexed rows from the given dataset. What is the value in the 3rd indexed column of the 332th indexed row in the selected dataframe. (row/ column indexed from zero(0)) [NAT]
- Please consider case sensitivity while giving your solution capital letter $\neq$ small letter
- If it is a string don't write quotes in the solution.


Ans : 352940 (Range: 352939 - 352941)

In [ ]:
df.iloc[1::2].reset_index(drop=True).iloc[332,3]

352940.0

#Q4 [Marks: 2] Select all the odd indexed columns from the given dataset. What is the value in the 5th indexed column of the 100th indexed row in the selected dataframe. (row/ column indexed from zero(0)) [MCQ]
- Please consider case sensitivity while giving your solution capital letter $\neq$ small letter
- If it is a string don't write quotes in the solution.

A) Single Family

B) West

C) 2022

D) Detached House

Ans: B

In [ ]:
df.iloc[:, 1::2].iloc[100,5]

'West'

#Q5 [Marks: 2] Select all the even indexed columns and even indexed rows from the given dataset. What is the value in the 3th indexed column of the 255th indexed row in the selected dataframe. (row/ column indexed/starts from zero(0)) [MCQ]
- Please consider case sensitivity while giving your solution capital letter $\neq$ small letter
- If it is a string don't write quotes in the solution.

A) Single Family

B) West

C) 2022

D) Detached House

Ans: D

In [ ]:
df.iloc[::2, ::2].reset_index(drop=True).iloc[255,3]

'Detached House'

#Q6 [Marks: 2] Determine the number of samples (rows) in the dataset that belong to the six most recent years, based on the provided data. [NAT]

Ans: 4749

In [ ]:
df['Year'].unique()

array([2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019,
       2020, 2021, 2022])

In [ ]:
df[df['Year'].isin([2017, 2018, 2019,2020, 2021, 2022])].shape

(4749, 12)

#Q7 [Marks: 2] Determine the number of samples (rows) in the dataset which has `num_rooms = 3` and `num_bathrooms = 3` [NAT]

Ans 2730

In [ ]:
df[(df['num_rooms'] == 3)&(df['num_bathrooms']== 3)].shape

(2730, 12)

#Q8 [Marks: 2] Determine the number of samples (rows) in the dataset which has `num_rooms = 3` or `num_bathrooms = 3` [NAT]

Ans 8591

In [ ]:
df[(df['num_rooms'] == 3)|(df['num_bathrooms']== 3)].shape

(8591, 12)

#Q9 [marks:2] In the **Year** 2022, how many houses (rows) located in the Greenwich **Locality** have exactly **num_room** == 3, and **facing** towards either the North or East? [NAT]

Ans: 42

In [ ]:
df[(df['Year']==2022)&(df['Locality'] == 'Greenwich')&(df['num_rooms']==3)&(df['Face'].isin(['North','East']))].shape[0]

42

#Q10[Marks: 2] 17 Calculate the number of samples(rows) that correspond to the month of August across all the years.? [NAT]
Ans: 1050

In [ ]:
df[pd.to_datetime(df['Date']).dt.month == 8].shape

(1050, 12)

#Q11 [Marks: 2] According to the data which `Locality` have highest average `Sale Price`? [MCQ]

A) Waterbury

B) Bridgeport

C) Greenwich

D) Norwalk

E) Fairfield

F) West Hartford

G) Stamford

Ans: C


In [ ]:
df[['Sale Price','Locality']].groupby(['Locality']).mean()

Sale Price
Locality                   
Bridgeport     1.740953e+05
Fairfield      7.277225e+05
Greenwich      2.270224e+06
Norwalk        5.590959e+05
Stamford       6.305313e+05
Waterbury      1.062152e+05
West Hartford  3.385472e+05